### PR TITLE
Add: Firebase Auth (Google, Facebook, Email)

### PR DESCRIPTION
**Pull request by saurabhchirde**
Features added
1. Users can log in using Google and Facebook accounts directly or by Email and password.
2. User can reset his password by providing a registered email-Id.

auth

### CODE REVIEW

File: src/Context/Network/NetworkContext.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  const { authDispatch } = useAuth();
+  const { alertDispatch } = useAlert();
+
+  const firebaseAuth = getAuth(app);
+  const provider = new GoogleAuthProvider();
+
+  const googleLoginHandler = async () => {
+    try {
+      const response = await signInWithPopup(firebaseAuth, provider);
+      console.log(response.user);
```
Comment (m26unkwn) [inline src/Context/Network/NetworkContext.jsx]: remove this console.log

**Review by m26unkwn — APPROVED** [review]
code is clean and ui is supperb
you can remove the unwanted console.log

File: src/Components/UI/Modal/Login.css [hunk 2]
```diff
@@ -2,6 +2,10 @@
   z-index: 4;
```
Comment (PuneetP16) [inline src/Components/UI/Modal/Login.css:2]: Suggestion: you could use variables for the z-indexes for ease of management.

File: src/Context/Network/NetworkContext.jsx [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+const NetworkContext = createContext({});
+
+const NetworkProvider = ({ children }) => {
+  const { authDispatch } = useAuth();
+  const { alertDispatch } = useAlert();
+  const { setShowLogin, setShowSignup } = useModal();
+
+  const googleProvider = new GoogleAuthProvider();
+  const facebookProvider = new FacebookAuthProvider();
+  // console.log(firebaseConfig);
```
Comment (PuneetP16) [inline src/Context/Network/NetworkContext.jsx]: Suggestion: remove console.log's is no longer required.

**Review by PuneetP16 — APPROVED** [review]
Your UI is awesome and functionality is working as expected. Good Overall work. Check the comments added. Looks good to merge.

**Pull request merged**